cmd/gcpNameResolver.go

```diff
@@ -0,0 +1,161 @@
+package cmd
+
+import (
+	"bytes"
+	"fmt"
+	"strconv"
+	"strings"
+)
+
+//For GCS, container naming follows:
+//1. Bucket names must contain only lowercase letters, numbers, dashes (-), underscores (_), and dots (.).
+//Spaces are not allowed. Names containing dots require verification.
+//2. Bucket names must start and end with a number or letter.
+//3. Bucket names must contain 3-63 characters. Names containing dots can contain up to 222 characters,
+//but each dot-separated component can be no longer than 63 characters.
+//4. Bucket names cannot be represented as an IP address in dotted-decimal notation (for example, 192.168.5.4).
+//5. Bucket names cannot begin with the "goog" prefix.
+//6. Bucket names cannot contain "google" or close misspellings, such as "g00gle".
+//For Azure, container/filesystem/share's naming follows:
+//1. Lower case letters, numbers and hyphen.
+//2. 3-63 length.
+//3. Name should not contain two consecutive hyphens.
+//4. Name should not start or end with hyphen.
+//In general we only have to fix the following cases:
+//1. Names with underscores (_). Here we will try to replace it with hyphens (-).
+// e.g. bucket_name -> bucket-name
+//2. Names with consecutive hyphens. Here we will replace it with -[number of hyphens]-.
+// e.g. bucket-----name -> bucket-5-name.
+//3. Names with period (.). Here we will replace it with hyphen (-).
+//4. In case of any naming collisions we try to add a suffix.
+// e.g. Bucket names: bucket-name, bucket.name. Then we will resolve into bucket.name -> bucket-name -> bucket-name-2
+
+type GCPBucketNameToAzureResourcesResolver struct {
+	bucketNameResolvingMap map[string]string
```

Comment:
I'm curious-- What causes us to require a second bucket resolver? How are S3 and GCS incompatible in their naming schemes that might cause this?

Reply:
GCS names contain underscores as well (_).